x-element.js

```diff
@@ -705,8 +716,9 @@ export default class XElement extends HTMLElement {
         invalidProperties.add(property);
       }
       hostInfo.initialized = true;
-      XElement.#updateHost(host);
```

Comment:
It turned out that more nuanced handling was conceptually needed here. I had initially done the broad-brush-stroke thing and just swapped the lines in `connectedCallback` — however, I think it makes more sense to just _delay_ this `#updateHost` call until after we've set up our listeners.

You can see that there are some minor gymnastics involved now, but I think this is the most-correct thing we can do here.

Reply:
Makes sense to me — thanks for all the helpful context, made review a breeze!